spec/models/scheduled_event_spec.rb

```diff
@@ -56,4 +57,61 @@
   	  expect(scheduled_event1.start_time).to eq Date.strptime(value, "%m/%d/%Y").to_date
   	end
   end
+   
+   context "Calender Event" do
+    let(:schedule_event_with_empty_recurring_rules) { FactoryGirl.create(:scheduled_event, :empty_recurring_rules, :offset_3)}
+    let(:scheduled_event_recurring_rules) { FactoryGirl.create(:scheduled_event, :empty_recurring_rules, :offset_3)}
+
+    it "should get self event" do
+      array = schedule_event_with_empty_recurring_rules.calendar_events(schedule_event_with_empty_recurring_rules.start_time, 4)
+      expect(array.length).to eq 1
+      expect(array).to eq [schedule_event_with_empty_recurring_rules]
+    end
+
+    it "should get 3 occurences" do
+      array = scheduled_event_recurring_rules.calendar_events(scheduled_event_recurring_rules.start_time, 4)
+      expect(array.length).to eq 1
+      expect(array.collect(&:start_time)).to eq [scheduled_event_recurring_rules.start_time]
+    end
+
+    it "should not reschedule for weekday for any offset" do
+      friday_offset_1 = FactoryGirl.create(:scheduled_event, :offset_1, :start_on_friday)
+      events = friday_offset_1.calendar_events(friday_offset_1.start_time, friday_offset_1.offset_rule)
+      expect(events.first.start_time.strftime('%A')).to eq 'Friday'
+    end
+
+    it "should reschedule to Tuesday for Sunday scheduled Event with offset 2" do
+      friday_offset_2 = FactoryGirl.create(:scheduled_event, :offset_2, :start_on_friday)
+      events = friday_offset_2.calendar_events(friday_offset_2.start_time, friday_offset_2.offset_rule)
+      expect(events.first.start_time.strftime('%A')).to eq 'Friday'
+    end
+
+    it "should reschedule to Monday for Sunday scheduled Event with offset 1" do
+      sunday_offset_1 = FactoryGirl.create(:scheduled_event, :offset_1, :start_on_sunday)
+      sunday_offset_1.update_attributes(recurring_rules: "{\"interval\":1,\"until\":null,\"count\":null,\"validations\":{\"day_of_week\":{},\"day_of_month\":[22]},\"rule_type\":\"IceCube::MonthlyRule\",\"week_start\":0}")
```

Comment:
@nks2109 put it in let variable the  attribute  value and you can access  which makes easy to look.